src/Auth/Login.jsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import "./index.css";
+import axios from "axios";
+class cbsLogin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      login: "show",
+      register: "hide",
+      boardingHouseNames: ["A.J", "Saigol", "Sanobar", "Mehran", "Abaseen"],
+      registerationDetails: {
+        name: "",
+        email: "",
+        password: "",
+        yearOfJoining: "",
+        boardingHouse: "",
+        phone: "",
+        username: ""
+      }
+    };
+
+    this.onChangeHandler = this.onChangeHandler.bind(this);
+    this.signUp = this.signUp.bind(this);
+    this.login = this.login.bind(this);
+  }
+  async signUp(ev) {
+    try {
+      ev.preventDefault();
+      const self = this;
+      var res = await axios({
+        method: "Post",
+        url: `${process.env.API_SERVER_URL}isUsernameUnique`,
+        data: {
+          newUser: self.state.registerationDetails
+        }
+      });
+      if (!res.data.isUnique) {
+        alert("Email Already Register");
+      }
+      this.showLogin();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+  async login(ev) {
+    try {
+      ev.preventDefault();
+      const self = this;
+      var res = await axios({
+        method: "Post",
+        url: `${process.env.API_SERVER_URL}/authenticateLogin`,
+        data: {
+          username: self.state.registerationDetails.email,
+          password: self.state.registerationDetails.password
+        }
+      });
+      if (!res.data.member.memberId) {
+        console.log("username/password error");
+      }
+      this.props.history.push("/profile");
+      console.log("login successfully");
+    } catch (err) {
+      alert("email/Password incorrect");
+    }
+  }
+  onChangeHandler(event) {
+    var tempRegisterationDetails = { ...this.state.registerationDetails };
+    tempRegisterationDetails[event.target.name] = event.target.value;
+    this.setState({ registerationDetails: tempRegisterationDetails });
+  }
+
+  showLogin() {
+    this.setState({ login: "show", register: "hide" });
+  }
+  showRegister() {
+    this.setState({ login: "hide", register: "show" });
+  }
+  render() {
+    var curentYear = new Date().getFullYear("yyyy");
```

Comment:
@Alihaider168 let's keep this computation out of render function so that it gets done only once and not on every call to render.